proposals/000-content-iterator.md

```diff
@@ -0,0 +1,223 @@
+# Content Iterator
+
+* Author: [Mickaël Menu](https://github.com/mickael-menu)
+* Review PR: [#177](https://github.com/readium/architecture/pull/177)
+
+## Summary
+
+A new Publication Service to iterate through a publication's content extracted as semantic elements.
+
+
+## Motivation
+
+The Content Iterator service provides a building block for many high-level features requiring access to the raw content of a publication, such as:
+
+* Text to speech
+* Accessibility readers
+* Basic search
+* Full-text search indexing
+* Image or audio indexes
+
+Today, implementing such features is complex because you need to:
+
+1. (maybe) Find a starting location in one of the reading order resources.
+2. Iterate through the reading order, opening and closing resources when needed.
+3. Extracting the textual or media content from the raw resource, which is different for every supported media type.
+
+The Content Iterator handles all that in a media type agnostic way.
+
+## Developer Guide
+
+### Iterating the content of a publication
+
+First, check if a `Publication` can be iterated using `publication.isContentIterable`.
+
+Then, you can request a `ContentIterator` from the publication:
+
+```typescript
+let iterator = publication.contentIteratorFrom(locator)
+if (iterator) {
+    ...
+}
+```
+
+The starting `Locator` is optional. When missing, the iteration will start from the very beginning of the publication.
+
+
+Once you have a valid `ContentIterator` instance, you can crawl through the publication content in both directions, using `previous` or `next`.
+
+```typescript
+var content: Content?
+while (content = iterator.next()) {
+    ...
+}
+
+iterator.close()
+```
+
+These APIs will return the `Content` elements in the reading order until it reaches the end (or the beginning for `previous`) of the publication. In which case, any additional calls will return `null`.
+
+:warning: Don't forget to `close` the iterator when you are done, to discard opened resources.
+
+### Extracting the data from `Content` elements
+
+The `Content` elements are value objects containing:
```

Comment:
Some new elements to take into account:

* SVG
    * can have an `aria-label` or `title` child element
* MathML
    * extracting the `tt` child elements could be useful
* MathJax